src/mongocxx/private/collection.hpp

```diff
@@ -39,6 +39,13 @@ class collection::impl {
           client_impl(client) {
     }
 
+    impl(const impl& i) : collection_t{libmongoc::collection_copy(i.collection_t)} { }
```

Comment:
we could potentially expose it somehow in bsoncxx or do it when you create a mongocxx::instance. I think I would find it confusing if I called set_new_handler, but the driver still allocated memory through raw malloc/free internally.